activity/src/main/java/com/amazonaws/stepfunctions/cloudformation/activity/CallbackContext.java

```diff
@@ -3,14 +3,16 @@
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
 import software.amazon.cloudformation.proxy.StdCallbackContext;
 
+@EqualsAndHashCode(callSuper = true)
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 @Builder
 public class CallbackContext extends StdCallbackContext {
-    private boolean isActivityDeletionStarted;
-    private int retryCount;
+    @Builder.Default
+    private boolean propagationDelayDone = false;
```

Comment:
Isn't this backwards incompatible? As this deploys, there is a short period where the `CallbackContext` of inflight deletion requests will not match the updated handler

Reply:
That's a good callout... I'm not sure if this would be backwards incompatible.

Will test locally to see what's the behaviour of the following scenario:

1. Old Delete handler is invoked
2. Old Delete handler returns IN_PROGRESS (with old CallbackContext)
3. Start waiting for 60s
4. New Delete handler takes effect
5. Finish waiting for 60s
6. New Delete handler is invoked with old CallbackContext <--- Not entirely sure what happens here when the old CallbackContext still has attributes `isActivityDeletionStarted` & `retryCount`, but the new CallbackContext doesn't will manually inject inputs to verify the behaviour

[Local testing using SAM AWS CLI](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-walkthrough.html#resource-type-walkthrough-test)

Reply:
Using the following sam-tests files:

```
{
  "action": "CREATE",
  "request": {
    "clientRequestToken": "4b90a7e4-b790-456b-a937-0cfdfa211dfe",
    "desiredResourceState": {
      "Name": "MyActivity-123"
    },
    "logicalResourceIdentifier": "MyActivity"
  },
  "callbackContext": ...,
  "credentials": {...}
}
```

```
{
  "action": "DELETE",
  "request": {
    "clientRequestToken": "4b90a7e4-b790-456b-a937-0cfdfa211dfd",
    "desiredResourceState": {
      "Name": "MyActivity-123",
      "Arn":"arn:aws:states:ca-central-1:XXXXXXXXXXXX:activity:MyActivity-123"
    },
    "logicalResourceIdentifier": "MyActivity"
  },
  "callbackContext": ...,
  "credentials": {...}
}
```

I've verified using AWS SAM CLI that the following callbackContexts are accepted (running `sam local invoke TestEntrypoint --event sam-tests/delete.json --region ca-central-1` after creating the resource):

Old delete handler (with dynamic stabilization) successfully executes with the following callbackContexts (enters the same code path as if `callbackContext == null` or `isActivityDeletionStarted = false`):
* `"callbackContext":{"callGraphs":{},"propagationDelayDone":true}`
* `"callbackContext":{"callGraphs":{},"propagationDelayDone":false}`

New delete handler (with static stabilization) successfully executes with the following callbackContexts (enters the same codepath as if `isPropagationDelayDone = false`):
* `"callbackContext":{"callGraphs":{},"retryCount":1,"activityDeletionStarted":true}`
* `"callbackContext":{"callGraphs":{},"retryCount":2,"activityDeletionStarted":true}`
* `"callbackContext":{"callGraphs":{},"retryCount":3,"activityDeletionStarted":true}`
* `"callbackContext":{"callGraphs":{},"retryCount":3,"activityDeletionStarted":true}`

This confirms your hypothesis and we shouldn't encounter any backwards incompatible behaviour.